### ISSUE TITLE
How to abstract across AudioContext and OfflineAudioContext?

### ISSUE THREAD

**Issue by swiftcoder** [rustwasm/wasm-bindgen#2738]
### Summary

In (untyped) Javascript it is relatively easy to write a function that accepts either AudioContext or OfflineAudioContext, and performs a set of audio operations thereon. However, in their web_sys versions, these two don't share any sort of trait for the common audio context operations. One can cast them both to BaseAudioContext, but this causes a big deprecation warning about Safari not supporting BaseAudioContext.

Is BaseAudioContext actually deprecated? I can't find any mention of this outside of the web_sys documentation. Is there another approach to accomplishing polymorphism here?

**Comment by jonboj** [comment]
Yes - confusing. In https://caniuse.com/?search=BaseAudioContext the note 1 states "_The BaseAudioContext interface itself is not present, but many of the methods are available on the AudioContext and OfflineAudioContext interfaces._"

In web-sys api both extends from `BaseAudioContext` [OfflineAudioContext](https://docs.rs/web-sys/0.3.55/src/web_sys/features/gen_OfflineAudioContext.rs.html#6), [AudioContext](https://docs.rs/web-sys/0.3.55/src/web_sys/features/gen_AudioContext.rs.html#6)

One level above [BaseAudioContext](https://docs.rs/web-sys/0.3.55/src/web_sys/features/gen_BaseAudioContext.rs.html#6) extends `js-sys::Object`, so guess that on Rust wasm side it should work fine to cast a `AudioContext / OfflineAudioContext` to `BaseAudioContext` and use this common interface.

**Comment by swiftcoder** [comment]
I'll give it a try.

That page also implies that this was fixed in almost all browsers over the last year - would it be reasonable to retire that deprecation annotation? I don't really want my build full of deprecation warnings :)
